docs/proposals/20200804-windows-support.md

```diff
@@ -0,0 +1,271 @@
+- Provide community guidance and scripts for building base images with the use of Windows nodes
+- Re-use of the existing Cluster API Bootstrap Provider Kubeadm and other tools where appropriate
+
+### Non-Goals/Future Work
+
+- Provide a way to run [control plane](https://cluster-api.sigs.k8s.io/reference/glossary.html#control-plane) nodes as Windows
+- Support for Windows versions outside of the Kubernetes support versions
+- Support for Windows nodes on the [management](https://cluster-api.sigs.k8s.io/reference/glossary.html#management-cluster) or [bootstrap clusters](https://cluster-api.sigs.k8s.io/reference/glossary.html#bootstrap-cluster)
+
+## Proposal
+
+There are several components that need to be addressed for Windows support:
+
+- [Cluster API Bootstrap Provider Kubeadm](https://github.com/kubernetes-sigs/cluster-api/tree/master/bootstrap/kubeadm)
+- VHD Image creation 
+- Kubelet and other component configuration
+- Infrastructure provider implementation 
+
+### Cluster API Bootstrap Provider Kubeadm
+
+#### cloud-init and cloudbase-init
+For Linux, the bootstrap script is provided to the infrastructure provider as a cloud-init script. 
+The infrastructure provider is responsible for putting the cloud-init script in the right location. 
+When the VM is booted, the cloud-init script runs automatically. 
+
+Cloud-init does not have Windows support. An alternative product is [cloudbase-init](https://github.com/cloudbase/cloudbase-init). 
+Cloudbase-init functions in the same way as cloud-init and can consume cloud-init scripts as provided by the Cluster API Bootstrap Provider Kubeadm.  
+By using cloudbase-init, Windows can leverage the existing solutions and stay up to date with the latest changes in CABPK.
+
+#### VHD Creation
+Using cloudbase-init requires the creation of a VHD with the tooling installed on it since it is not 
+provided out of the box by any cloud providers.  To address this we will provide packer scripts as part of 
+the [image-builder project](https://github.com/kubernetes-sigs/image-builder) that will pre-install 
+cloudbase init.  It is important to note that while scripts can be provided to build a  VHD.  All VHD’s 
+built will need to adhere to [Windows licensing requirements](https://www.microsoft.com/en-us/licensing/product-licensing/windows-server).
+
+AKS-Engine has an example implementation for using packer and scripts to do VHD configuration: https://github.com/Azure/aks-engine/blob/master/vhd/packer/windows-vhd-builder.json
+
+Although the Linux implementation in image-builder uses Ansible for configuration, Windows will not share 
+the configuration. Therefore, Ansible is not a requirement for Windows image builder implementation.  
+The main requirement for VHD building is that the build can be started from a Linux or Windows machine.
+
+#### Kubelet and other component configuration
+Due to the lack of privileged containers in Windows, a combination of `PreKubeadmCommands`/`PostKubeadmCommands` 
```

Comment:
Can you expand on why PreKubeadmCommands/PostKubeadmCommands are needed here? Why not put everything in the bootstrap script?

Reply:
nvm just read the end of the paragraph that talks about this